lib/repository/app_repository.dart

```diff
@@ -32,9 +34,19 @@ class AppRepository extends DemoDecoratorComposite {
     if (backup != null) {
       return _pair(qrMessage, userId, backup);
     } else {
-      return _pair(qrMessage, userId) //
-          .then((_) => _sdk.startKeygen().value)
-          .then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);
+      return _pair(qrMessage, userId).then((_) async {
```

Comment:
Let's extract this handler into helper keygen method - high level pairing logic now is interspersed with lower-level tracking logic.

Reply:
You mean sth like this:
```dart

    if (backup != null) {
      return _pair(qrMessage, userId, backup);
    } else {
      return _pair(qrMessage, userId).then((_) async {
        return await _keygen();
      }).then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);
    }
  }

  Future<Keyshare2> _keygen() async {
    try {
      _analyticManager.trackDistributedKeyGen(type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.initiated);
      final keyshare = await _sdk.startKeygen().value;
      _analyticManager.trackDistributedKeyGen(
          type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.success, publicKey: keyshare.ethAddress);
      return keyshare;
    } catch (error) {
      _analyticManager.trackDistributedKeyGen(
          type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.failed, error: error.toString());
      rethrow;
    }
  }
  ```

Reply:
Yes, but do why do we need to use await before _keygen call?
```suggestion
return _pair(qrMessage, userId) //
  .then((_) => _keygen())
  .then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);```

Reply:
We can't